Shell/main.cpp

```diff
@@ -56,6 +56,31 @@ void FileDescriptionCollector::add(int fd)
     m_fds.append(fd);
 }
 
+SavedFileDescriptors::SavedFileDescriptors(const NonnullRefPtrVector<AST::Rewiring>& intended_rewirings)
+{
+    for (auto& rewiring : intended_rewirings) {
+        int new_fd = fcntl(0, F_DUPFD, rewiring.source_fd);
```

Comment:
This is one weird interface...
`fcntl` already takes an fd as its first arg, _why_ would anyone think "hey, let's just ignore the perfectly fine fd argument and ask for a third argument for `F_DUPFD`"? hm Dr. POSIX? hm?

Reply:
Hmmm...this needs `CLOEXEC`

Reply:
Uh what's wrong with using `dup()` here?

Reply:
Nothing wrong with `dup()`, I just wanted to whine about how weird `fcntl` is lol